childhood/vulnerability/ive-junaeb.ipynb

```diff
@@ -0,0 +1,683 @@
+      "470714         0     32  \n",
+      "470715         0      0  \n",
+      "470716         0      2  \n",
+      "470717         0      1  \n",
+      "470718         0      2  \n",
+      "470719         0      2  \n",
+      "470720         0      1  \n",
+      "470721         0     35  \n",
+      "470722         0      1  \n",
+      "470723         0     14  \n",
+      "470724         0      1  \n",
+      "470725         0      5  \n",
+      "470726         0      1  \n",
+      "470727         0      2  \n",
+      "470728         0     12  \n",
+      "470729         0      5  \n",
+      "\n",
+      "[470730 rows x 8 columns]\n",
+      "104.54 s | Exported csv file.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from bs4 import BeautifulSoup\n",
+    "import requests\n",
+    "import pandas as pd\n",
+    "import urllib.request\n",
+    "from urllib.parse import urlparse\n",
+    "import time\n",
+    "import os\n",
+    "import re\n",
+    "import urllib3\n",
+    "\n",
+    "# declares timer and locks absolute path to working directory\n",
+    "before = time.perf_counter()\n",
+    "absolute_path = os.getcwd()\n",
+    "\n",
+    "# solves SSL certificate issues error when retrieving files \n",
+    "import os, ssl\n",
+    "if (not os.environ.get(\"PYTHONHTTPSVERIFY\", \"\") and getattr(ssl, \"_create_unverified_context\", None)): \n",
+    "    ssl._create_default_https_context = ssl._create_unverified_context\n",
+    "\n",
+    "# Solving SSL certificate issue\n",
+    "urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)\n",
+    "\n",
+    "# retrieve links from site\n",
+    "url = \"https://www.junaeb.cl/ive\"\n",
+    "page = requests.get(url, verify=False)\n",
+    "soup = BeautifulSoup(page.text, \"lxml\")\n",
+    "a_tags = soup.find_all('a')\n",
+    "all_links = [link.get('href') for link in a_tags]\n",
+    "links = []\n",
+    "for link in all_links:\n",
+    "    if \".xls\" in link and \"IVESINAE_COMUNA_2013\" not in link:\n",
+    "        links.append(link)\n",
+    "        \n",
+    "years_df = []\n",
+    "\n",
+    "# creates data_temp folder and changes working directory\n",
+    "if os.path.isdir(\"data_temp\") == False:\n",
+    "    os.mkdir(\"data_temp\")    \n",
+    "os.chdir(\"data_temp\")\n",
+    "\n",
+    "# downloads files\n",
+    "for url in links:\n",
+    "    encoded_url = urllib.parse.quote(url.encode('utf-8'),':/')\n",
+    "    filename = encoded_url[encoded_url.rfind(\"/\")+1:]\n",
+    "    if filename not in os.listdir():\n",
+    "        urllib.request.urlretrieve(encoded_url, filename)\n",
+    "        print(\"{:.2f} s | Downloaded {}\".format(time.perf_counter()-before, url))\n",
+    "    \n",
+    "# processing function\n",
+    "def clean(file, year):\n",
+    "    \n",
+    "    #defines scope of variable\n",
+    "    global years_df\n",
+    "        \n",
+    "    # reads Excel file and defines available sheets on file\n",
+    "    try:\n",
```

Comment:
Use `regexp` instead of `try` for to test `BASICA|BÁSICA` words

Reply:
I found a new way to check the Excel file's sheet names and selecting them without try/except, it's more efficient now.